src/OrchardCore.Modules/OrchardCore.Security/Extensions/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using OrchardCore.Security;
+using OrchardCore.Security.Middlewares;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public static class ApplicationBuilderExtensions
+    {
+        public static IApplicationBuilder UseSecurityHeaders(this IApplicationBuilder app)
+        {
+            if (app is null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            return app.UseSecurityHeaders(config => { });
+        }
+
+        public static IApplicationBuilder UseSecurityHeaders(this IApplicationBuilder app, Action<SecurityHeadersBuilder> configureSecurityHeaders)
+        {
+            if (app is null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            if (configureSecurityHeaders is null)
+            {
+                throw new ArgumentNullException(nameof(configureSecurityHeaders));
+            }
+
+            var settings = new SecuritySettings();
+            var builder = new SecurityHeadersBuilder(settings);
+
+            configureSecurityHeaders.Invoke(builder);
+
+            settings = builder.Build();
+
+            app.UseMiddleware<ReferrerPolicyMiddleware>(settings.ReferrerPolicy);
```

Comment:
Check if we can aggregate all the middlewares into `SecurityHeadersMiddleware`

Reply:
I decided to create a module per header for custom use, the we can create `SecurityHeadersMiddleware` for the module purpose